File: packages/dd-trace/test/llmobs/parity/scenarios/openai-agents/guardrails.js
'use strict'

const { guardrailAgent, setup } = require('./common')

async function main () {
  const { agents, finish } = setup(true)
  await agents.run(guardrailAgent(), 'What is the sum of 1 and 2?')
  await finish()
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})

File: packages/dd-trace/test/llmobs/parity/scenarios/openai-agents/guardrails.py
import asyncio

from agents import Runner

from _agents_common import finish
from _agents_common import guardrail_agent
from _agents_common import setup


async def main():
    setup(chat_api=True)
    await Runner.run(guardrail_agent(), "What is the sum of 1 and 2?")
    finish()


asyncio.run(main())
